### PR TITLE
YBRO 3 cards

### PR DESCRIPTION
**Pull request by Simisays**
Tested them and they seem to work.

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/lonely_end.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Name:Lonely End
+ManaCost:1 B
+Types:Instant
+A:SP$ Charm | Choices$ DBDebuff,DBLoyalty | CharmNum$ 1
+SVar:DBDebuff:DB$ Pump | ValidTgts$ Creature | NumAtt$ -3 | NumDef$ -3 | SubAbility$ DBGainLife | SpellDescription$ Target creature gets -3/-3 until end of turn
+SVar:DBLoyalty:DB$ RemoveCounter | ValidTgts$ Planeswalker | CounterType$ LOYALTY | CounterNum$ 3 | SubAbility$ DBGainLife | SpellDescription$ Remove three loyalty counters from target planeswalker
+SVar:DBGainLife:DB$ GainLife | LifeAmount$ 2 | CheckSVar$ X | SVarCompare$ EQ0 | SpellDescription$ If you weren't the starting player, you gain two life 
+SVar:X:Count$StartingPlayer.1.0
+DeckHas:Ability$LifeGain
+Oracle:Choose one -\n* Target creature gets -3/-3 until end of turn\n*Remove three loyalty counters from target planeswalker\nIf you weren't the starting player, you gain 2 life.
```
Comment (paulsnoops) [inline forge-gui/res/cardsfolder/upcoming/lonely_end.txt:10]: * should be •
Also, need a space on the second one before "Remove"

File: forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+Name:Argivian Welcome
+ManaCost:W U
+Types:Instant
+A:SP$ Destroy | ValidTgts$ Creature.powerGE4 | TargetMin$ 0 | TargetMax$ 1 | TgtPrompt$ Select up to one target creature with power 4 or greater | SubAbility$ DBChoose | SpellDescription$ Destroy up to one target creature with power 4 or greater. Choose a nonland card in your hand. It perpetually gains flash. 
+SVar:DBChoose:DB$ ChooseCard | ChoiceZone$ Hand | Choices$ Card.nonLand+YouOwn | ChoiceTitle$ Choose a nonland card in your hand | Amount$ 1 | SubAbility$ DBEffect
+SVar:DBEffect:DB$ Effect | StaticAbilities$ PerpetualFlash| Name$ Argivian Welcome's Perpetual Effect | Duration$ Permanent | SubAbility$ DBCleanup
```
Comment (paulsnoops) [inline forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt:6]: PerpetualFlash|
needs a space

File: forge-gui/res/cardsfolder/upcoming/crucias_titan_of_the_waves.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+SVar:TrigDiscard:DB$ Discard | Defined$ You | NumCards$ 1 | Mode$ TgtChoose | RememberDiscarded$ True | SubAbility$ DBChoose
+SVar:DBChoose:DB$ GenericChoice | Choices$ Ambitious,Expedient | Defined$ You | AILogic$ Random | ConditionPresent$ Card | ConditionCompare$ GE1
+SVar:Ambitious:DB$ ChangeZone | Origin$ Library | Destination$ Hand | AtRandom$ True | NoShuffle$ True | Mandatory$ True | NoLooking$ True | NoReveal$ True | ChangeType$ Card.cmcGTX | ChangeNum$ 1 | SubAbility$ DBCleanup
+SVar:Expedient:DB$ ChangeZone | Origin$ Library | Destination$ Hand | AtRandom$ True | NoShuffle$ True | Mandatory$ True | NoLooking$ True | NoReveal$ True | ChangeType$ Card.cmcLTX | ChangeNum$ 1 | SubAbility$ DBCleanup
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+DeckHas:Ability$Token|Discard & Type$Treasure
```
Comment (paulsnoops) [inline forge-gui/res/cardsfolder/upcoming/crucias_titan_of_the_waves.txt:11]: DeckHas:Ability$Sacrifice|Token|Discard & Type$Treasure|Artifact

File: forge-gui/res/cardsfolder/upcoming/crucias_titan_of_the_waves.txt [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+SVar:X:Remembered$CardManaCost
+Oracle:At the beginning of your end step, you may discard a card. If you do, create a Treasure token and choose ambitious or expedient. If you chose ambitious, seek a card with greater mana value than the discarded card. If you chose expedient, seek a card with lesser mana value than the discarded card.
```
Comment (paulsnoops) [inline forge-gui/res/cardsfolder/upcoming/crucias_titan_of_the_waves.txt:13]: There should be a token creation before the choose?

File: forge-gui/res/cardsfolder/upcoming/crucias_titan_of_the_waves.txt [hunk 5]
```diff
@@ -0,0 +1,14 @@
+Name:Crucias, Titan of the Waves
+ManaCost:1 B R
+Types:Legendary Creature Human Pirate
+PT:3/3
+T:Mode$ Phase | Phase$ End of Turn | OptionalDecider$ You | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigDiscard | TriggerDescription$ At the beginning of your end step, you may discard a card. If you do, create a Treasure token and choose ambitious or expedient. If you chose ambitious, seek a card with greater mana value than the discarded card. If you chose expedient, seek a card with lesser mana value than the discarded card. 
+SVar:TrigDiscard:DB$ Discard | Defined$ You | NumCards$ 1 | Mode$ TgtChoose | RememberDiscarded$ True | SubAbility$ DBToken
+SVar:DBToken:DB$ Token | TokenScript$ c_a_treasure_sac | ConditionPresent$ Card | ConditionCompare$ GE1 | SubAbility$ DBChoose
+SVar:DBChoose:DB$ GenericChoice | ConditionPresent$ Card | ConditionCompare$ GE1 | Choices$ Ambitious,Expedient | Defined$ You | AILogic$ Random
+SVar:Ambitious:DB$ ChangeZone | Origin$ Library | Destination$ Hand | AtRandom$ True | NoShuffle$ True | Mandatory$ True | NoLooking$ True | NoReveal$ True | ChangeType$ Card.cmcGTX | ChangeNum$ 1 | SubAbility$ DBCleanup
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/crucias_titan_of_the_waves.txt:9]: How is the GUI for this? I think lines 9 and 10 will need SpellDescription to properly populate the GenericChoice popup menu?

File: forge-gui/res/cardsfolder/upcoming/lonely_end.txt [hunk 6]
```diff
@@ -0,0 +1,10 @@
+Name:Lonely End
+ManaCost:1 B
+Types:Instant
+A:SP$ Charm | Choices$ DBDebuff,DBLoyalty | CharmNum$ 1
+SVar:DBDebuff:DB$ Pump | ValidTgts$ Creature | NumAtt$ -3 | NumDef$ -3 | SubAbility$ DBGainLife | SpellDescription$ Target creature gets -3/-3 until end of turn
+SVar:DBLoyalty:DB$ RemoveCounter | ValidTgts$ Planeswalker | CounterType$ LOYALTY | CounterNum$ 3 | SubAbility$ DBGainLife | SpellDescription$ Remove three loyalty counters from target planeswalker
+SVar:DBGainLife:DB$ GainLife | LifeAmount$ 2 | CheckSVar$ X | SVarCompare$ EQ0 | SpellDescription$ If you weren't the starting player, you gain two life
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/lonely_end.txt:7]: punctuate Desc

File: forge-gui/res/cardsfolder/upcoming/lonely_end.txt [hunk 7]
```diff
@@ -0,0 +1,10 @@
+Name:Lonely End
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/lonely_end.txt:1]: missing periods from many Descs and the Oracle

File: forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt [hunk 8]
```diff
@@ -0,0 +1,9 @@
+Name:Argivian Welcome
+ManaCost:W U
+Types:Instant
+A:SP$ Destroy | ValidTgts$ Creature.powerGE4 | TargetMin$ 0 | TargetMax$ 1 | TgtPrompt$ Select up to one target creature with power 4 or greater | SubAbility$ DBChoose | SpellDescription$ Destroy up to one target creature with power 4 or greater. Choose a nonland card in your hand. It perpetually gains flash. 
+SVar:DBChoose:DB$ ChooseCard | ChoiceZone$ Hand | Choices$ Card.nonLand+YouOwn | ChoiceTitle$ Choose a nonland card in your hand | Amount$ 1 | SubAbility$ DBEffect
+SVar:DBEffect:DB$ Effect | StaticAbilities$ PerpetualFlash | Name$ Argivian Welcome's Perpetual Effect | Duration$ Permanent | SubAbility$ DBCleanup
+SVar:PerpetualLifelink:Mode$ Continuous | Affected$ Card.ChosenCard | AddKeyword$ Flash | EffectZone$ Command | AffectedZone$ Battlefield,Hand,Graveyard,Exile,Stack,Library,Command | Description$ The chosen card perpetually gains flash.
+SVar:DBCleanup:DB$ Cleanup | ClearChosenCard$ True
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt:8]: You may not need Cleanup... try it without and see?

File: forge-gui/res/cardsfolder/upcoming/crucias_titan_of_the_waves.txt [hunk 9]
```diff
@@ -0,0 +1,14 @@
+Name:Crucias, Titan of the Waves
+ManaCost:1 B R
+Types:Legendary Creature Human Pirate
+PT:3/3
+T:Mode$ Phase | Phase$ End of Turn | OptionalDecider$ You | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigDiscard | TriggerDescription$ At the beginning of your end step, you may discard a card. If you do, create a Treasure token and choose ambitious or expedient. If you chose ambitious, seek a card with greater mana value than the discarded card. If you chose expedient, seek a card with lesser mana value than the discarded card. 
+SVar:TrigDiscard:DB$ Discard | Defined$ You | NumCards$ 1 | Mode$ TgtChoose | RememberDiscarded$ True | SubAbility$ DBToken
+SVar:DBToken:DB$ Token | TokenScript$ c_a_treasure_sac | ConditionPresent$ Card | ConditionCompare$ GE1 | SubAbility$ DBChoose
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/crucias_titan_of_the_waves.txt:7]: Is this really working in both cases without `ConditionDefined$` ???

File: forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt (hunk 8 above)
Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt:8]: OK, thanks for checking. I need to look into that code anyway

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt:8]: Actually the card doesn't work at all. Does flash have to added as a static with CastWithFlash  ?

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt:8]: nvm, the problem was that perpetuallifelink needed to be perpetualflash :z

**Review by paulsnoops — DISMISSED** [review]

File: forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt [hunk 10]
```diff
@@ -0,0 +1,8 @@
+Name:Argivian Welcome
+ManaCost:W U
+Types:Instant
+A:SP$ Destroy | ValidTgts$ Creature.powerGE4 | TargetMin$ 0 | TargetMax$ 1 | TgtPrompt$ Select up to one target creature with power 4 or greater | SubAbility$ DBChoose | SpellDescription$ Destroy up to one target creature with power 4 or greater. Choose a nonland card in your hand. It perpetually gains flash.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt:4]: how does the stack description look when you cast this?

File: forge-gui/res/cardsfolder/upcoming/crucias_titan_of_the_waves.txt [hunk 11]
```diff
@@ -0,0 +1,14 @@
+Name:Crucias, Titan of the Waves
+ManaCost:1 B R
+Types:Legendary Creature Human Pirate
+PT:3/3
+T:Mode$ Phase | Phase$ End of Turn | OptionalDecider$ You | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigDiscard | TriggerDescription$ At the beginning of your end step, you may discard a card. If you do, create a Treasure token and choose ambitious or expedient. If you chose ambitious, seek a card with greater mana value than the discarded card. If you chose expedient, seek a card with lesser mana value than the discarded card. 
+SVar:TrigDiscard:DB$ Discard | Defined$ You | NumCards$ 1 | Mode$ TgtChoose | RememberDiscarded$ True | SubAbility$ DBToken
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/crucias_titan_of_the_waves.txt:6]: make the discard a Cost - then you should be able to use `SVar:X:Discarded$CardManaCost` like Mercurial Chemister

you should also be able to eliminate Remembered/Cleanup line if you do this

File: forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt (hunk 10 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt:4]: image

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt:4]: I guess Stackdescription$ spelldescription would be better ?

File: forge-gui/res/cardsfolder/upcoming/crucias_titan_of_the_waves.txt [hunk 12]
```diff
@@ -3,7 +3,7 @@ ManaCost:1 B R
 Types:Legendary Creature Human Pirate
 PT:3/3
 T:Mode$ Phase | Phase$ End of Turn | OptionalDecider$ You | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigToken | TriggerDescription$ At the beginning of your end step, you may discard a card. If you do, create a Treasure token and choose ambitious or expedient. If you chose ambitious, seek a card with greater mana value than the discarded card. If you chose expedient, seek a card with lesser mana value than the discarded card. 
-SVar:TrigToken:AB$ Token | Cost$ Discard<1/Card> |  TokenScript$ c_a_treasure_sac | SubAbility$ DBChoose
+SVar:TrigToken:AB$ Token | Cost$ Discard<1/Card> | TokenScript$ c_a_treasure_sac | SubAbility$ DBChoose
 SVar:DBChoose:DB$ GenericChoice | Choices$ Ambitious,Expedient | Defined$ You | AILogic$ Random
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/crucias_titan_of_the_waves.txt:7]: How does the GenericChoice dialogue box look?

I feel like line 8 and 9 SpellDescription should start with Ambitious and Expedient

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/crucias_titan_of_the_waves.txt:7]: Hmm good point

File: forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt [hunk 13]
```diff
@@ -0,0 +1,8 @@
+Name:Argivian Welcome
+ManaCost:W U
+Types:Instant
+A:SP$ Destroy | ValidTgts$ Creature.powerGE4 | TargetMin$ 0 | TargetMax$ 1 | TgtPrompt$ Select up to one target creature with power 4 or greater | SubAbility$ DBChoose | StackDescription$ SpellDescription | SpellDescription$ Destroy up to one target creature with power 4 or greater. Choose a nonland card in your hand. It perpetually gains flash. 
+SVar:DBChoose:DB$ ChooseCard | ChoiceZone$ Hand | Choices$ Card.nonLand+YouOwn | ChoiceTitle$ Choose a nonland card in your hand | Amount$ 1 | SubAbility$ DBEffect
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt:5]: `Mandatory$ True`

File: forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt [hunk 14]
```diff
@@ -0,0 +1,8 @@
+Name:Argivian Welcome
+ManaCost:W U
+Types:Instant
+A:SP$ Destroy | ValidTgts$ Creature.powerGE4 | TargetMin$ 0 | TargetMax$ 1 | TgtPrompt$ Select up to one target creature with power 4 or greater | SubAbility$ DBChoose | StackDescription$ SpellDescription | SpellDescription$ Destroy up to one target creature with power 4 or greater. Choose a nonland card in your hand. It perpetually gains flash.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt:4]: If you do it this way you will still get unwanted StackDesc from line 5

File: forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt [hunk 15]
```diff
@@ -1,8 +1,8 @@
 Name:Argivian Welcome
 ManaCost:W U
 Types:Instant
-A:SP$ Destroy | ValidTgts$ Creature.powerGE4 | TargetMin$ 0 | TargetMax$ 1 | TgtPrompt$ Select up to one target creature with power 4 or greater | SubAbility$ DBChoose | StackDescription$ SpellDescription | SpellDescription$ Destroy up to one target creature with power 4 or greater. Choose a nonland card in your hand. It perpetually gains flash. 
-SVar:DBChoose:DB$ ChooseCard | ChoiceZone$ Hand | Choices$ Card.nonLand+YouOwn | ChoiceTitle$ Choose a nonland card in your hand | Amount$ 1 | SubAbility$ DBEffect
+A:SP$ Destroy | ValidTgts$ Creature.powerGE4 | TargetMin$ 0 | TargetMax$ 1 | TgtPrompt$ Select up to one target creature with power 4 or greater | SubAbility$ DBChoose | StackDescription$  SpellDescription$ Destroy up to one target creature with power 4 or greater. Choose a nonland card in your hand. | SpellDescription$ Destroy up to one target creature with power 4 or greater. Choose a nonland card in your hand. It perpetually gains flash.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt:4]: `StackDescription$  SpellDescription$` is not what you want

File: forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt [hunk 16]
```diff
@@ -0,0 +1,8 @@
+Name:Argivian Welcome
+ManaCost:W U
+Types:Instant
+A:SP$ Destroy | ValidTgts$ Creature.powerGE4 | TargetMin$ 0 | TargetMax$ 1 | TgtPrompt$ Select up to one target creature with power 4 or greater | SubAbility$ DBChoose | StackDescription$ Destroy up to one target creature with power 4 or greater. Choose a nonland card in your hand. | SpellDescription$ Destroy up to one target creature with power 4 or greater. Choose a nonland card in your hand. It perpetually gains flash.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/argivian_welcome.txt:4]: That's still wrong/redundant: `StackDescription$ SpellDescription`

**Review by tool4ever — APPROVED** [review]

**Pull request merged**